Sources/SWBUtil/ProcessController.swift

```diff
@@ -0,0 +1,177 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public import Foundation
+
+#if canImport(Subprocess)
+import Subprocess
+#endif
+
+import Synchronization
+
+#if canImport(System)
+public import System
+#else
+public import SystemPackage
+#endif
+
+public final class ProcessController: Sendable {
+    public let path: Path
+    public let arguments: [String]
+    public let environment: Environment?
+    public let workingDirectory: Path?
+    private let state = SWBMutex<State>(.unstarted)
+    private let done = WaitCondition()
+
+    private struct RunningState {
+        var task: Task<Void, Never>
+        var pid: pid_t?
+    }
+
+    private enum State {
+        case unstarted
+        case running(_ runningState: RunningState)
+        case exited(exitStatus: Result<Processes.ExitStatus, any Error>)
+    }
+
+    public init(path: Path, arguments: [String], environment: Environment?, workingDirectory: Path?) {
+        self.path = path
+        self.arguments = arguments
+        self.environment = environment
+        self.workingDirectory = workingDirectory
+    }
+
+    public func start(input: FileDescriptor, output: FileDescriptor, error: FileDescriptor, highPriority: Bool) {
+        state.withLock { state in
+            guard case .unstarted = state else {
+                fatalError("API misuse: process was already started")
+            }
+
+            func updateState(processIdentifier pid: pid_t?) {
+                self.state.withLock { state in
+                    guard case var .running(runningState) = state, runningState.pid == nil else {
+                        preconditionFailure() // unreachable
+                    }
+                    runningState.pid = pid
+                    state = .running(runningState)
+                }
+            }
+
+            let task = Task<Void, Never>.detached { [path, arguments, environment, workingDirectory, done] in
+                defer { done.signal() }
+                let result = await Result.catching {
+                    #if !canImport(Darwin) || os(macOS)
+                    #if canImport(Subprocess)
+                    var platformOptions = PlatformOptions()
+                    platformOptions.teardownSequence = [.gracefulShutDown(allowedDurationToNextStep: .seconds(5))]
+                    #if os(macOS)
+                    if highPriority {
+                        platformOptions.qualityOfService = .userInitiated
```

Comment:
@compnerd Would it seem reasonable to call SetPriorityClass with HIGH_PRIORITY_CLASS for Windows (REALTIME_PRIORITY_CLASS definitely sounded excessive)? Currently this is used to elevate the build service process so that it can serve high-priority requests more efficiently (e.g. with Swift Concurrency task priority set higher than normal).